### PR TITLE
ISPN-12725 Conflict resolution fails in transactional cache with auto-commit disabled

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-12725
https://issues.redhat.com/browse/ISPN-12715
https://issues.redhat.com/browse/ISPN-12717

ISPN-12725 Conflict resolution fails in transactional cache with auto-commit disabled
ISPN-12715 Conflict resolution failure logged at debug level
ISPN-12717 DefaultConflictManager.getAllVersions fails in tx caches

### CODE REVIEW

**Review by pruivo — APPROVED** [review]

**Comment by danberindei** [comment]
@pruivo updated to include the IllegalArgumentException fix

**Comment by danberindei** [comment]
Updated to fix the test failures (sometimes the transaction manager is null) and to rebase on master.

File: core/src/test/java/org/infinispan/conflict/impl/TxConflictResolutionTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ * 1. do a split, and let k -> A, k -> null in the two partitions
+ * 2. merge partitions
+ * 3. check that all members read A
+ * </p>
+ *
+ * <p>See ISPN-12725.</p>
+ *
+ * @author Ryan Emerson
+ * @since 9.1
+ */
```
Comment (ryanemerson) [inline core/src/test/java/org/infinispan/conflict/impl/TxConflictResolutionTest.java:25]: I think this needs updating or removed :slightly_smiling_face:

**Review by ryanemerson — APPROVED** [review]
@danberindei Other than the javadoc, LGTM!

File: core/src/test/java/org/infinispan/conflict/impl/TxConflictResolutionTest.java (hunk 1 above)
Comment (danberindei) [reply core/src/test/java/org/infinispan/conflict/impl/TxConflictResolutionTest.java:27]: Updated

**Comment by ryanemerson** [comment]
Thanks @danberindei

**Pull request merged**